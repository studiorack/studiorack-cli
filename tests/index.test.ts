import { expect, test } from 'vitest';
import { cleanOutput, cliCatch } from './shared';
import { RegistryType } from '@open-audio-stack/core';

test('Root command', async () => {
  const error = cliCatch();
  expect(error.exitCode).toBe(1);
  expect(cleanOutput(error.stderr)).toMatchSnapshot();
});

test('Root command plugins', async () => {
  const error = cliCatch(RegistryType.Plugins);
  expect(error.exitCode).toBe(1);
  expect(cleanOutput(error.stderr)).toMatchSnapshot();
});

test('Root command presets', async () => {
  const error = cliCatch(RegistryType.Presets);
  expect(error.exitCode).toBe(1);
  expect(cleanOutput(error.stderr)).toMatchSnapshot();
});

test('Root command projects', async () => {
  const error = cliCatch(RegistryType.Projects);
  expect(error.exitCode).toBe(1);
  expect(cleanOutput(error.stderr)).toMatchSnapshot();
});
