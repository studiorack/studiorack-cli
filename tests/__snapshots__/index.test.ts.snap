// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Root command 1`] = `
Usage: index [options] [command]

Options:
  -V, --version   output the version number
  -h, --help      display help for command

Commands:
  config          View/update configuration
  plugins
  presets
  projects
  help [command]  display help for command
`;

exports[`Root command plugins 1`] = `
Usage: index plugins [options] [command]

Options:
  -h, --help                        display help for command

Commands:
  create [options] <path>           Create a new package locally
  filter [options] <field> <value>  Filter the by field and matching value
  get [options] <input>             Get package metadata from registry
  install [options] <input>         Install a package locally by slug/version
  list [options]                    List packages
  reset [options]                   Reset the synced package cache
  scan [options]                    Scan local packages into cache
  search [options] <query>          Search using a lazy matching query
  sync [options]                    Sync remote packages into cache
  uninstall [options] <input>       Uninstall a package locally by slug/version
  help [command]                    display help for command
`;

exports[`Root command presets 1`] = `
Usage: index presets [options] [command]

Options:
  -h, --help                        display help for command

Commands:
  create [options] <path>           Create a new package locally
  filter [options] <field> <value>  Filter the by field and matching value
  get [options] <input>             Get package metadata from registry
  install [options] <input>         Install a package locally by slug/version
  list [options]                    List packages
  reset [options]                   Reset the synced package cache
  scan [options]                    Scan local packages into cache
  search [options] <query>          Search using a lazy matching query
  sync [options]                    Sync remote packages into cache
  uninstall [options] <input>       Uninstall a package locally by slug/version
  help [command]                    display help for command
`;

exports[`Root command projects 1`] = `
Usage: index projects [options] [command]

Options:
  -h, --help                        display help for command

Commands:
  create [options] <path>           Create a new package locally
  filter [options] <field> <value>  Filter the by field and matching value
  get [options] <input>             Get package metadata from registry
  install [options] <input>         Install a package locally by slug/version
  list [options]                    List packages
  reset [options]                   Reset the synced package cache
  scan [options]                    Scan local packages into cache
  search [options] <query>          Search using a lazy matching query
  sync [options]                    Sync remote packages into cache
  uninstall [options] <input>       Uninstall a package locally by slug/version
  help [command]                    display help for command
`;
